keycloak/event-listeners/src/main/java/org/eclipse/sw360/keycloak/event/listener/service/Sw360UserService.java

```diff
@@ -10,109 +10,372 @@
+            return users;
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            logger.error("Event listener error retrieving all users from SW360 database", e);
+            return Collections.emptyList();
         }
     }
 
+    /**
+     * Retrieves a user by email address.
+     * 
+     * @param email the email address to search for
+     * @return User if found, null otherwise
+     */
     public User getUserByEmail(String email) {
+        if (email == null || email.trim().isEmpty()) {
+            logger.warn("Event listener attempted to get user with null or empty email");
+            return null;
+        }
+        
         try {
-            UserService.Iface sw360UserClient = getThriftUserClient();
-            return sw360UserClient.getByEmail(email);
+            logger.debug("Event listener retrieving user by email: {}", email);
+            User user = connector.get(User.class, email);
+            if (user != null) {
+                logger.debug("Event listener found user for email: {}", email);
+            } else {
+                logger.debug("Event listener found no user for email: {}", email);
+            }
+            return user;
         } catch (Exception e) {
-            throw new RuntimeException(e);
+            logger.error("Event listener error retrieving user by email: " + email, e);
+            return null;
         }
     }
 
+    /**
+     * Retrieves a user by email or external ID.
+     * First attempts to find by email, then searches all users by external ID.
+     * 
+     * @param userIdentifier the email or external ID to search for
+     * @return User if found, null otherwise
+     */
     public User getUserByEmailOrExternalId(String userIdentifier) {
+        if (userIdentifier == null || userIdentifier.trim().isEmpty()) {
+            logger.warn("Event listener attempted to get user with null or empty identifier");
+            return null;
+        }
+        
         try {
-            UserService.Iface sw360UserClient = getThriftUserClient();
-            return sw360UserClient.getByEmailOrExternalId(userIdentifier, userIdentifier);
+            logger.debug("Event listener retrieving user by email or external ID: {}", userIdentifier);
+            
+            // First try by email/ID
+            User user = connector.get(User.class, userIdentifier);
+            if (user != null) {
+                logger.debug("Event listener found user by direct lookup for identifier: {}", userIdentifier);
+                return user;
+            }
+            
+            // If not found, search by external ID
+            logger.debug("Event listener searching by external ID for identifier: {}", userIdentifier);
+            List<User> allUsers = connector.getAll(User.class);
+            Optional<User> foundUser = allUsers.stream()
+                .filter(u -> userIdentifier.equals(u.getExternalid()))
+                .findFirst();
```

Comment:
This can be optimized by not getting all users and querying `byEmail` and `byExternalId` views instead.

https://github.com/eclipse-sw360/sw360/blob/558851af121fc0a423deeca4c4b0822e3f7a0af1/backend/common/src/main/java/org/eclipse/sw360/datahandler/db/UserRepository.java#L162-L174